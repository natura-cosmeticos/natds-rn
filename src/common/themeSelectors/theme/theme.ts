import { tokens as styleTokens } from '@naturacosmeticos/natds-styles';
import { IColorThemeTokens } from '@naturacosmeticos/natds-styles/dist/tokens/colors/themeBound/IColorThemeTokens';
import { IElevation } from '@naturacosmeticos/natds-styles/dist/tokens/elevation/IElevation';

type Tokens = typeof styleTokens;
type TokensWithoutColor = Omit<Tokens, 'colors'>
type TokensWithoutColorTokens = Omit<TokensWithoutColor, 'colorTokens'>
type TokensWithoutElevation = Omit<TokensWithoutColorTokens, 'elevation'>

interface ElevationProps {
  elevation: number
  shadowColor: string
  shadowOffset: { height: number, width: number }
  shadowOpacity: number
  shadowRadius: number
}

type Elevation = Record<keyof IElevation, ElevationProps>


export type Theme = TokensWithoutElevation
  & {
    colorTokens: IColorThemeTokens,
    elevation: Elevation | {}
  }

type ThemeMode = 'dark' | 'light';

const parseElevation = (elevation: string) => {
  const regEx = /(\d+(\.\d+)?)/g;
  const matches = elevation.match(regEx);

  if (matches) {
    const [width, height, blurRadius, , r, g, b, alpha] = matches;

    const toUnit = (prop: string) => parseInt(prop, 0);

    return {
      elevation: 2,
      shadowColor: `rgb(${r}, ${g}, ${b})`,
      shadowOffset: { height: toUnit(height), width: toUnit(width) },
      shadowOpacity: parseFloat(alpha),
      shadowRadius: toUnit(blurRadius),
    };
  }

  return {};
};

const buidElevation = (elevation: IElevation) => {
  const nativeElevation = {};

  Object
    .keys(elevation)
    .forEach((key) => { nativeElevation[key] = parseElevation(elevation[key]); });

  return nativeElevation;
};

export const buildTheme = (tokens: Tokens, brand: string, mode: ThemeMode) => {
  const colorThemeName = `${brand}${mode === 'light' ? 'Light' : 'Dark'}ColorTokens`;
  const localTokens = { ...tokens };

  delete localTokens.colors;

  const theme: Theme = {
    ...localTokens,
    colorTokens: localTokens.colorTokens[colorThemeName],
    elevation: buidElevation(localTokens.elevation),
  };

  return theme;
};

interface ThemeSelector {
  (theme: Theme): string | undefined
}

export const getTheme = (selector: ThemeSelector) => (
  { theme }: { theme: Theme },
) => selector(theme);
