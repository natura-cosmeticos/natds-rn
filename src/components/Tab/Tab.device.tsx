import { storiesOf } from '@storybook/react-native'
import {
  Default, Variants, Interactive
} from './Tab.stories'

storiesOf('Tab', module)
  .add('Default', Default)
  .add('Variants', Variants)
  .add('Interactive', Interactive)
