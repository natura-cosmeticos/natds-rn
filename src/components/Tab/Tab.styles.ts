import styled from 'styled-components/native';
import {
  getColorMediumEmphasis,
  getColorOnBackground,
  getColorSecondary,
  getColorSurface,
  getFont,
  getSpacingSmall,
  getShadowBySize,
  Theme,
} from '../../common/themeSelectors';

export type TabButtonTypes = 'primary' | 'secondary';

interface TabBase {
  type: TabButtonTypes
  theme: Theme
}

const getTabButtonStyles = (theme: Theme, type: TabButtonTypes) => {
  const styles = {
    primary: {
      borderBottomColor: getColorSecondary(theme),
    },
    secondary: {
      borderColor: 'transparent',
    },
  };

  return styles[type];
};

const getTabTextStyles = (theme: Theme, type: TabButtonTypes) => {
  const styles = {
    primary: {
      color: getColorOnBackground(theme),
    },
    secondary: {
      color: getColorMediumEmphasis(theme),
    },
  };

  return styles[type];
};

export const TabWrapper = styled.View(({ theme }) => ({
  border: 'none',
  flexDirection: 'row',
  ...getShadowBySize(theme, '6'),
}));

export const TabButton = styled.TouchableOpacity<TabBase>(({ type, theme }) => ({
  alignItems: 'center',
  backgroundColor: getColorSurface(theme),
  border: 'transparent',
  borderStyle: 'solid',
  borderWidth: 2,
  flex: 1,
  padding: getSpacingSmall(theme),
  ...getTabButtonStyles(theme, type),
}));

export const TabText = styled.Text<TabBase>(({ type, theme }) => ({
  fontFamily: getFont('bold'),
  fontSize: 14,
  letterSpacing: 1.25,
  ...getTabTextStyles(theme, type),
}));
