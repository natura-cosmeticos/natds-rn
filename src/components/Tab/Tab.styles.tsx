/* eslint-disable max-lines */
import styled from 'styled-components/native';
import { AccessibilityState } from 'react-native';

import {
  getColorMediumEmphasis,
  getColorLowEmphasis,
  getColorOnBackground,
  getColorPrimary,
  getColorSurface,
  getSpacingSmall,
  getSizeLargeX,
  getSizeMedium,
  getShadowBySize,
  Theme,
} from '../../common/themeSelectors';
import {
  TabButtonTypes, TabProps, IconPositions,
} from './Tab.types';
import { TouchableRipple } from '../TouchableRipple/TouchableRipple';

interface TabBase {
  type: TabButtonTypes
  theme: Theme
  accessibilityLabel?: string
  accessibilityHint?: string
  accessibilityState?: AccessibilityState
  accessible?: boolean
}

interface TextBase extends TabBase {
  iconPosition?: IconPositions
  disabled?: boolean
}

interface DisplayText {
  theme: Theme
  iconPosition?: IconPositions
}

interface TabWrapperProp {
  theme: Theme
  elevation?: boolean
  backgroundColor?: boolean
  iconPosition?: IconPositions
}

interface ButtonContentProps extends TabProps {
  theme: Theme,
}

const getTabButtonStyles = (theme: Theme, type: TabButtonTypes) => {
  const styles = {
    primary: {
      borderBottomColor: getColorPrimary(theme),
    },
    secondary: {
      borderColor: 'transparent',
    },
  };

  return styles[type];
};

export const getTabTextStyles = (theme: Theme, type: TabButtonTypes) => {
  const styles = {
    primary: getColorOnBackground(theme),
    secondary: getColorMediumEmphasis(theme),
  };

  return styles[type];
};

export const getDisplayTextStyles = ({ theme, iconPosition }: DisplayText) => {
  if (iconPosition) {
    const styles = {
      icon: {
        display: 'none',
      },
      leading: {
        paddingLeft: theme.spacing.micro,
      },
      top: {
        paddingTop: theme.spacing.micro,
      },
    };

    return styles[iconPosition];
  }

  return {};
};

export const getTabTextColor = ({ disabled, theme, type }: TextBase) => {
  if (disabled) {
    return { color: getColorLowEmphasis(theme) };
  }

  return { color: getTabTextStyles(theme, type) };
};

export const getTabWrapperElevation = ({ elevation, theme }: TabWrapperProp) => (
  elevation && getShadowBySize(theme, 'micro')
);

export const TabWrapper = styled.View.attrs({
  horizontal: true,
  showsHorizontalScrollIndicator: false,
})<TabWrapperProp>(({
  theme, backgroundColor, iconPosition,
}) => ({
  backgroundColor: backgroundColor ? getColorSurface(theme) : 'transparent',
  border: 'none',
  flexDirection: 'row',
  height: iconPosition === 'icon' ? getSizeMedium(theme) : getSizeLargeX(theme),
}));


export const TabButton = styled(TouchableRipple)<TabBase>(({ type, theme }) => ({
  ...getTabButtonStyles(theme, type),
  alignItems: 'center',
  borderBottomWidth: 2,
  borderStyle: 'solid',
  flex: 1,
  justifyContent: 'center',
  minWidth: getSizeLargeX(theme),
  paddingHorizontal: getSpacingSmall(theme),
}));

export const TabText = styled.Text<TextBase>(({
  type, theme, iconPosition, disabled = false,
}) => ({
  ...getDisplayTextStyles({ iconPosition, theme }),
  ...getTabTextColor({ disabled, theme, type }),
  fontFamily: theme.tab.label.primary.fontFamily,
  fontSize: theme.tab.label.fontSize,
  fontWeight: theme.tab.label.primary.fontWeight,
  letterSpacing: theme.tab.label.letterSpacing,
  lineHeight: theme.tab.label.fontSize * theme.tab.label.lineHeight,
}));

export const TabButtonContent = styled.View<Pick<ButtonContentProps, 'iconPosition'>>(({ iconPosition }) => ({
  alignItems: 'center',
  flexDirection: iconPosition === 'leading' ? 'row' : 'column',
}));
