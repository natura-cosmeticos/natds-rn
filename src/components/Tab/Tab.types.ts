/* eslint-disable max-len */
import { AccessibilityRole } from 'react-native'
import { Theme } from '../../common/themeSelectors'

export type TabButtonTypes = 'primary' | 'secondary'
export type IconPositions = 'leading' | 'top' | 'icon'
export type TabPositions = 'fixed' | 'scrollable'
export interface TabOptions {
  disabled?: boolean;
  iconName?: string;
  key: string;
  label: string;
  selected?: boolean;
}

export interface TabProps {
  /**
   * Communicates the purpose of a component to the user of an assistive technology.
   * @default 'tab'
   */
   accessibilityRole?: AccessibilityRole;
  /**
   * When `true`, indicates that the view is an accessibility element.
   * @default false
   */
  accessible?: boolean;
  /**
   * Provide a background color to the tab.
   * @default true
   */
  backgroundColor?: boolean;
  /**
   * Elevation of the tab, applies a shadow to the tab.
   * @default false
   */
  elevation?: boolean;
  /**
   * Icon position relative to text label `leading` | `top` | `icon`.
   * @default 'none'
   */
  iconPosition?: IconPositions;
  /**
   * Tab position relative to screen `fixed` | `scrollable`.
   * @default 'fixed'
   */
  position?: TabPositions;
  /**
   * Array of Tab options with `key`, `label`, `selected`, `disabled`, and `iconName` properties.
   */
  tabOptions: TabOptions[];
  /**
   * The onChange prop allows passing a function that will be called
   * once the tab option has been pressed.
   */
  onChange: (value: number) => void;
  /**
   * The tab theme
   */
  theme: Theme;
  /**
   * Optional ID for testing
   */
  testID?: string;
}
