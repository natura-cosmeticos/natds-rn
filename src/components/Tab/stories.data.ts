export const defaultOptions = [
  {
    key: 'itemOne',
    label: 'Item One'
  },
  {
    key: 'itemTwo',
    label: 'Item Two'
  },
  {
    key: 'itemThree',
    label: 'Item Three'
  }
]

export const elevationOptions = [
  {
    key: 'itemOne',
    label: 'Item One'
  },
  {
    key: 'itemTwo',
    label: 'Item Two'
  },
  {
    key: 'itemThree',
    label: 'Item Three'
  }
]

export const disabledOptions = [
  {
    disabled: true,
    key: 'itemOne',
    label: 'Item One'
  },
  {
    key: 'itemTwo',
    label: 'Item Two',
    selected: true
  },
  {
    key: 'itemThree',
    label: 'Item Three'
  }
]

export const iconOptions = [
  {
    iconName: 'outlined-default-mockup',
    key: 'itemOne',
    label: 'Item One'
  },
  {
    iconName: 'outlined-default-mockup',
    key: 'itemTwo',
    label: 'Item Two'
  },
  {
    iconName: 'outlined-default-mockup',
    key: 'itemThree',
    label: 'Item Three'
  }
]

export const interactiveOptions = [
  {
    iconName: 'outlined-default-mockup',
    key: 'itemOne',
    label: 'Item One'
  },
  {
    iconName: 'outlined-default-mockup',
    key: 'itemTwo',
    label: 'Item Two'
  },
  {
    iconName: 'outlined-default-mockup',
    key: 'itemThree',
    label: 'Item Three'
  },
  {
    iconName: 'outlined-default-mockup',
    key: 'itemFourth',
    label: 'Item Four'
  }
]
