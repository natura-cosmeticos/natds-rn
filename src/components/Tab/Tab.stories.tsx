/* eslint-disable max-lines */
import React from 'react'
import { View } from 'react-native'

import { boolean, select } from '@storybook/addon-knobs'
import { Tab } from '.'
import { TabPositions, IconPositions } from './Tab.types'

const description = () => `
- - -

  ### NOTE:
  This component is available in the following variants:

  - ✅ Standard

With the following attribute status:

  - **Position:**
    - ✅ \`Fixed\`
    - ✅ \`Scrollable\`
  - **Icon:**
    - ✅ \`Leading\`
    - ✅ \`Top\`
    - ✅ \`Icon\`
  - ✅ **Disabled**
  - ✅ **Elevation**
  - ✅ **Color**
  - ✅ **onChange**

- - -
`

export default {
  component: Tab,
  parameters: {
    componentSubtitle: 'Tabs allow the users to navigate between groups of contents that are at the same level of hierarchy.',
    docs: {
      extractComponentDescription: description
    }
  },
  title: 'Components|Tab'
}

const tabPositions = {
  fixed: 'fixed',
  scrollable: 'scrollable'
}

const iconPositions = {
  icon: 'icon',
  leading: 'leading',
  top: 'top'
}

export const Positions = () => {
  const positionTabOptions = [
    {
      key: 'itemOne',
      label: 'Item One'
    },
    {
      key: 'itemTwo',
      label: 'Item Two'
    },
    {
      key: 'itemThree',
      label: 'Item Three'
    }
  ]

  const scrollableTabOptions = [
    ...positionTabOptions,
    {
      key: 'itemFourth',
      label: 'Item Four'
    },
    {
      key: 'itemFifth',
      label: 'Item Five'
    }
  ]

  return (
    <>
      <View style={{ maxWidth: 600, paddingBottom: 30 }}>
        <Tab tabOptions={positionTabOptions} onChange={() => ({})} />
      </View>
      <View style={{ maxWidth: 600, paddingBottom: 30 }}>
        <Tab position="scrollable" tabOptions={scrollableTabOptions} onChange={() => ({})} />
      </View>
    </>
  )
}

export const Icon = () => {
  const iconTabOptions = [
    {
      iconName: 'outlined-default-mockup',
      key: 'itemOne',
      label: 'Item One'
    },
    {
      iconName: 'outlined-default-mockup',
      key: 'itemTwo',
      label: 'Item Two'
    },
    {
      iconName: 'outlined-default-mockup',
      key: 'itemThree',
      label: 'Item Three'
    }
  ]

  return (
    <>
      <View style={{ maxWidth: 600, paddingBottom: 30 }}>
        <Tab iconPosition="leading" tabOptions={iconTabOptions} onChange={() => ({})} />
      </View>
      <View style={{ maxWidth: 600, paddingBottom: 30 }}>
        <Tab iconPosition="top" tabOptions={iconTabOptions} onChange={() => ({})} />
      </View>
      <View style={{ maxWidth: 600, paddingBottom: 30 }}>
        <Tab iconPosition="icon" tabOptions={iconTabOptions} onChange={() => ({})} />
      </View>
    </>
  )
}

export const Disabled = () => {
  const tabOptions = [
    {
      disabled: true,
      key: 'itemOne',
      label: 'Item One'
    },
    {
      key: 'itemTwo',
      label: 'Item Two',
      selected: true
    },
    {
      key: 'itemThree',
      label: 'Item Three'
    }
  ]

  return (
    <View style={{ maxWidth: 600, paddingBottom: 30 }}>
      <Tab tabOptions={tabOptions} onChange={() => ({})} />
    </View>
  )
}

export const Elevation = () => {
  const tabOptions = [
    {
      key: 'itemOne',
      label: 'Item One'
    },
    {
      key: 'itemTwo',
      label: 'Item Two'
    },
    {
      key: 'itemThree',
      label: 'Item Three'
    }
  ]

  return (
    <View style={{ maxWidth: 600, paddingBottom: 30 }}>
      <Tab elevation tabOptions={tabOptions} onChange={() => ({})} />
    </View>
  )
}

export const Color = () => {
  const tabOptions = [
    {
      key: 'itemOne',
      label: 'Item One'
    },
    {
      key: 'itemTwo',
      label: 'Item Two'
    },
    {
      key: 'itemThree',
      label: 'Item Three'
    }
  ]

  return (
    <View style={{ maxWidth: 600, paddingBottom: 30 }}>
      <Tab backgroundColor={false} tabOptions={tabOptions} onChange={() => ({})} />
    </View>
  )
}

export const Interactive = () => {
  const interactiveTabOptions = [
    {
      iconName: 'outlined-default-mockup',
      key: 'itemOne',
      label: 'Item One'
    },
    {
      iconName: 'outlined-default-mockup',
      key: 'itemTwo',
      label: 'Item Two'
    },
    {
      iconName: 'outlined-default-mockup',
      key: 'itemThree',
      label: 'Item Three'
    },
    {
      iconName: 'outlined-default-mockup',
      key: 'itemFourth',
      label: 'Item Four'
    }
  ]

  return (
    <View style={{ maxWidth: 600, paddingBottom: 30 }}>
      <Tab
        tabOptions={interactiveTabOptions}
        onChange={() => ({})}
        elevation={boolean('Elevation', false)}
        backgroundColor={boolean('BackgroundColor', true)}
        position={select('Position', tabPositions, 'fixed') as TabPositions}
        iconPosition={select('Icon Position', iconPositions, 'icon') as IconPositions}
      />
    </View>
  )
}

export const All = () => (
  <>
    <Color />
    <Disabled />
    <Elevation />
    <Icon />
    <Positions />
  </>
)
