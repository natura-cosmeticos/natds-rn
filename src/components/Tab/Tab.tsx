/* eslint-disable max-len */
import { IconName } from '@naturacosmeticos/natds-icons'
import React, { useEffect, useState, useCallback } from 'react'
import { ScrollView, View } from 'react-native'
import { withTheme } from 'styled-components/native'

import { Icon } from '../Icon'
import {
  TabWrapper, TabButton, TabText, TabButtonContent, getTabTextColor, getTabWrapperElevation
} from './Tab.styles'
import { TabProps, TabButtonTypes } from './Tab.types'

const TabComponent = ({
  testID = 'ds-tab', theme, tabOptions, onChange, position = 'fixed',
  iconPosition, elevation = false, backgroundColor = true, accessible = false,
  accessibilityRole = 'tab'
}: TabProps) => {
  const [activeTab, setActiveTab] = useState(0)

  const TabPositionType = position === 'fixed' ? View : ScrollView

  const handlePress = (index: number) => {
    onChange(index)
    setActiveTab(index)
  }

  const getCurrentType = (index: number, disabled = false): TabButtonTypes => {
    if (activeTab === index && !disabled) {
      return 'primary'
    }

    return 'secondary'
  }

  const getSelectedTab = useCallback((): number => {
    let selected = 0

    tabOptions.find((tab, index) => {
      if (tab.selected && !tab.disabled) {
        selected = index

        return true
      }

      return false
    })

    return selected
  }, [tabOptions])

  useEffect(() => {
    const selectedIndex = getSelectedTab()

    setActiveTab(selectedIndex)
  }, [getSelectedTab])

  return (
    <View
      // This view is to apply the shadow when the position is scrollable in IOS, but not working in Android.
      style={getTabWrapperElevation({ elevation, theme })}
      accessible={accessible}
      accessibilityRole={accessibilityRole}
    >
      <TabWrapper
        // This view is to apply the shadow to working in Android.
        style={getTabWrapperElevation({ elevation, theme })}
        as={TabPositionType}
        iconPosition={iconPosition}
        backgroundColor={backgroundColor}
        testID={testID}
      >
        {tabOptions.map((tabOption, index) => {
          const currentType = getCurrentType(index, tabOption.disabled)

          return (
            <TabButton
              accessible={accessible}
              disabled={tabOption.disabled}
              key={tabOption.key}
              onPress={() => handlePress(index)}
              type={currentType}
              testID={`${testID}-item-${index}`}
            >
              <TabButtonContent iconPosition={iconPosition}>
                {tabOption.iconName && iconPosition
                  && (
                  <Icon
                    accessibilityRole="imagebutton"
                    style={getTabTextColor({ disabled: tabOption.disabled, theme, type: currentType })}
                    name={tabOption.iconName as IconName}
                    size="standard"
                  />
                  )}
                <TabText iconPosition={iconPosition} type={currentType} disabled={tabOption.disabled}>
                  {tabOption.label.toUpperCase()}
                </TabText>
              </TabButtonContent>
            </TabButton>
          )
        })}
      </TabWrapper>
    </View>
  )
}

export const Tab = withTheme(TabComponent)
