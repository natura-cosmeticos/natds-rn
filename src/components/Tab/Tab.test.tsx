import React from 'react';
import renderer from 'react-test-renderer';
import { fireEvent, render } from '@testing-library/react-native';
import '@testing-library/jest-native/extend-expect';
import { ThemeProvider } from 'styled-components/native';
import theme from '../../common/themeSelectors/theme/mock-theme.json';
import { Tab } from '.';
import { TabProps } from './Tab';

jest.mock('react-native/Libraries/Components/Touchable/TouchableOpacity',
  () => 'TouchableOpacity');

jest.mock('../../common/themeSelectors', () => (
  {
    getColorMediumEmphasis: () => '#777777',
    getColorOnBackground: () => '#333333',
    getColorPrimary: () => '#F4F4',
    getColorSurface: () => '#FFFFFF',
    getFont: () => 'doc robot',
    getShadowBySize: () => ({ shadowColor: '#AEAEAE' }),
    getSpacingSmall: () => 16,
  }));

const renderTab = (fn, props: Omit<TabProps, 'theme'>) => (fn(
  <ThemeProvider theme={theme}>
    <Tab {...props} />
  </ThemeProvider>,
));

const defaultProps = ({
  onChange: () => {},
  tabOptions: [
    {
      key: 'itemOne',
      label: 'Item One',
    },
    {
      key: 'itemTwo',
      label: 'Item Two',
    },
    {
      key: 'itemThree',
      label: 'Item Three',
    },
  ],
});

describe('Tab component', () => {
  it('Should call the given onChange function', () => {
    const onChangeMock = jest.fn();
    const { queryByTestId } = renderTab(render, {
      ...defaultProps,
      onChange: onChangeMock,
    });

    const secondTab = queryByTestId('tab').children[1];

    fireEvent.press(secondTab);

    expect(onChangeMock).toHaveBeenCalledTimes(1);
  });

  it('Should have bottom border on the first tab as default', () => {
    const { queryByTestId } = renderTab(render, defaultProps);

    const firstTab = queryByTestId('tab').children[0];

    expect(firstTab).toHaveStyle({
      borderBottomColor: '#F4F4',
    });
  });

  it('Should add bottom border on tab item when the given tab is pressed', () => {
    const { queryByTestId } = renderTab(render, defaultProps);

    const thirdTab = queryByTestId('tab').children[2];

    fireEvent.press(thirdTab);

    expect(thirdTab).toHaveStyle({
      borderBottomColor: '#F4F4',
    });
  });

  it('Should render tab item with a uppercase text', () => {
    const { queryByTestId } = renderTab(render, defaultProps);

    const firstTab = queryByTestId('tab').children[0];

    expect(firstTab).toHaveTextContent('ITEM ONE');
  });

  it('Should render tab component without passing any type prop', () => {
    const tab = renderTab(renderer.create, defaultProps).toJSON();

    expect(tab).toMatchSnapshot();
  });
});
