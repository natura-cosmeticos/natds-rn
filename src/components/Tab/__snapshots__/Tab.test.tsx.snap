// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tab component Should render tab component with default props 1`] = `
<View
  accessibilityRole="tab"
  accessible={false}
  style={false}
>
  <View
    backgroundColor={true}
    horizontal={true}
    showsHorizontalScrollIndicator={false}
    style={
      Array [
        Object {
          "backgroundColor": "#FFFFFF",
          "borderColor": "black",
          "borderStyle": "solid",
          "borderWidth": 0,
          "flexDirection": "row",
          "height": 72,
        },
        false,
      ]
    }
    testID="ds-tab"
  >
    <View
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderBottomColor": "#F4F4",
            "borderBottomWidth": 2,
            "borderStyle": "solid",
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "justifyContent": "center",
            "minWidth": 72,
            "paddingHorizontal": 16,
          },
        ]
      }
      testID="ds-tab-item-0"
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "column",
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#333333",
                "fontFamily": "Roboto",
                "fontSize": 14,
                "fontWeight": "500",
                "letterSpacing": 1.23,
                "lineHeight": 21,
              },
            ]
          }
          type="primary"
        >
          ITEM ONE
        </Text>
      </View>
    </View>
    <View
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderBottomWidth": 2,
            "borderColor": "transparent",
            "borderStyle": "solid",
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "justifyContent": "center",
            "minWidth": 72,
            "paddingHorizontal": 16,
          },
        ]
      }
      testID="ds-tab-item-1"
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "column",
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#777777",
                "fontFamily": "Roboto",
                "fontSize": 14,
                "fontWeight": "500",
                "letterSpacing": 1.23,
                "lineHeight": 21,
              },
            ]
          }
          type="secondary"
        >
          ITEM TWO
        </Text>
      </View>
    </View>
    <View
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderBottomWidth": 2,
            "borderColor": "transparent",
            "borderStyle": "solid",
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "justifyContent": "center",
            "minWidth": 72,
            "paddingHorizontal": 16,
          },
        ]
      }
      testID="ds-tab-item-2"
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "column",
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#777777",
                "fontFamily": "Roboto",
                "fontSize": 14,
                "fontWeight": "500",
                "letterSpacing": 1.23,
                "lineHeight": 21,
              },
            ]
          }
          type="secondary"
        >
          ITEM THREE
        </Text>
      </View>
    </View>
    <View
      accessibilityState={
        Object {
          "disabled": true,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderBottomWidth": 2,
            "borderColor": "transparent",
            "borderStyle": "solid",
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "justifyContent": "center",
            "minWidth": 72,
            "paddingHorizontal": 16,
          },
        ]
      }
      testID="ds-tab-item-3"
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "column",
            },
          ]
        }
      >
        <Text
          disabled={true}
          style={
            Array [
              Object {
                "color": "#BBBBBB",
                "fontFamily": "Roboto",
                "fontSize": 14,
                "fontWeight": "500",
                "letterSpacing": 1.23,
                "lineHeight": 21,
              },
            ]
          }
          type="secondary"
        >
          ITEM FOUR
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`Tab component render props Should render tab component with scrollable postion 1`] = `
<View
  accessibilityRole="tab"
  accessible={false}
  style={false}
>
  <RCTScrollView
    backgroundColor={true}
    horizontal={true}
    showsHorizontalScrollIndicator={false}
    style={
      Array [
        Object {
          "backgroundColor": "#FFFFFF",
          "borderColor": "black",
          "borderStyle": "solid",
          "borderWidth": 0,
          "flexDirection": "row",
          "height": 72,
        },
        false,
      ]
    }
    testID="ds-tab"
  >
    <View>
      <View
        accessibilityState={
          Object {
            "disabled": false,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "borderBottomColor": "#F4F4",
              "borderBottomWidth": 2,
              "borderStyle": "solid",
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "justifyContent": "center",
              "minWidth": 72,
              "paddingHorizontal": 16,
            },
          ]
        }
        testID="ds-tab-item-0"
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "column",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#333333",
                  "fontFamily": "Roboto",
                  "fontSize": 14,
                  "fontWeight": "500",
                  "letterSpacing": 1.23,
                  "lineHeight": 21,
                },
              ]
            }
            type="primary"
          >
            ITEM ONE
          </Text>
        </View>
      </View>
      <View
        accessibilityState={
          Object {
            "disabled": false,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "borderBottomWidth": 2,
              "borderColor": "transparent",
              "borderStyle": "solid",
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "justifyContent": "center",
              "minWidth": 72,
              "paddingHorizontal": 16,
            },
          ]
        }
        testID="ds-tab-item-1"
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "column",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#777777",
                  "fontFamily": "Roboto",
                  "fontSize": 14,
                  "fontWeight": "500",
                  "letterSpacing": 1.23,
                  "lineHeight": 21,
                },
              ]
            }
            type="secondary"
          >
            ITEM TWO
          </Text>
        </View>
      </View>
      <View
        accessibilityState={
          Object {
            "disabled": false,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "borderBottomWidth": 2,
              "borderColor": "transparent",
              "borderStyle": "solid",
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "justifyContent": "center",
              "minWidth": 72,
              "paddingHorizontal": 16,
            },
          ]
        }
        testID="ds-tab-item-2"
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "column",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#777777",
                  "fontFamily": "Roboto",
                  "fontSize": 14,
                  "fontWeight": "500",
                  "letterSpacing": 1.23,
                  "lineHeight": 21,
                },
              ]
            }
            type="secondary"
          >
            ITEM THREE
          </Text>
        </View>
      </View>
      <View
        accessibilityState={
          Object {
            "disabled": true,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "borderBottomWidth": 2,
              "borderColor": "transparent",
              "borderStyle": "solid",
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "justifyContent": "center",
              "minWidth": 72,
              "paddingHorizontal": 16,
            },
          ]
        }
        testID="ds-tab-item-3"
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "column",
              },
            ]
          }
        >
          <Text
            disabled={true}
            style={
              Array [
                Object {
                  "color": "#BBBBBB",
                  "fontFamily": "Roboto",
                  "fontSize": 14,
                  "fontWeight": "500",
                  "letterSpacing": 1.23,
                  "lineHeight": 21,
                },
              ]
            }
            type="secondary"
          >
            ITEM FOUR
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
